### PR TITLE
Upgrade Ember to 2.3 and refactor to remove deprecations.

### PR DESCRIPTION
**Pull request by westlywright**
This pr should be paired with https://github.com/rancher/ember-api-store/pull/7 to ensure that the ui has all the
upgrades and doesnt take a dump.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+---
```
Comment (vincent99) [inline .travis.yml]: Can you remove (not add) this file that ember-cli helpfully created, we don't use travis..

File: app/hosts/new/custom/controller.js [hunk 2]
```diff
@@ -2,35 +2,35 @@ import Ember from 'ember';
 import ManageLabels from 'ui/mixins/manage-labels';
 import Util from 'ui/utils/util';
 
+const DELAY = 2000;
```
Comment (vincent99) [inline app/hosts/new/custom/controller.js]: Unused?  I assume this is one of the jshint errors

File: package.json [hunk 3]
```diff
@@ -23,33 +23,38 @@
   "devDependencies": {
     "broccoli-asset-rev": "^2.2.0",
     "ember-api-store": "^1.2.13",
+    "ember-ajax": "0.7.1",
+    "ember-data": "^2.3.0",
```
Comment (vincent99) [inline package.json]: Are `ember-ajax` or `ember-data` actually needed?  I know tests used to require ember-data...

File: package.json [hunk 4]
```diff
@@ -23,33 +23,38 @@
...
     "ember-cli-inject-live-reload": "^1.3.1",
     "ember-cli-inline-content": "^0.4.0",
-    "ember-cli-qunit": "^1.0.4",
+    "ember-cli-qunit": "^1.2.1",
     "ember-cli-release": "0.2.8",
     "ember-cli-sass": "^5.1.0",
-    "ember-cli-sri": "^1.2.0",
+    "ember-cli-sri": "^2.0.0",
     "ember-cli-uglify": "^1.2.0",
+    "ember-data": "^2.3.0",
```
Comment (vincent99) [inline package.json]: Ok it's definitely not needed twice :smile:

Comment (westlywright) [inline package.json]: whoops think that happened in rebase, and i'll verify we can remove it

**Comment by vincent99** [comment]
ember-cli-build.js needs these removed:

```
  app.import('bower_components/zeroclipboard/dist/ZeroClipboard.js');
  app.import('bower_components/zeroclipboard/dist/ZeroClipboard.swf', {destDir: 'assets'});
```

**Comment by vincent99** [comment]
Note to future me: requires https://github.com/rancher/ember-api-store/pull/7

**Comment by vincent99** [comment]
@westlywright: `{{display-name-select}}` and `registry-credential-select` are broken, can probably just be replaced with the new-embery way of doing selects or `{{new-select}}`
image

Also needs rebase after other merges...

**Pull request merged**